### PR TITLE
test upload and download subsetting flags

### PR DESCRIPTION
**Pull request by ransomw1c**
add tests for `--files` flag on `bundle upload` and the `--name-filter` flag on `bundle download`

### CODE REVIEW

File: pkg/core/bundle_unpack.go [hunk 1]
```diff
@@ -466,6 +465,7 @@ func downloadBundleEntries(ctx context.Context, bundle *Bundle,
 		bundle.l.Info("downloading bundle entries",
 			zap.Int("num", len(bundle.BundleEntries)))
 		for _, b := range bundle.BundleEntries {
+			var selectionPredicateOk bool
```
Comment (ransomw1c) [inline pkg/core/bundle_unpack.go]: at first tests were failing b/c i didn't notice that `z` matches `leafsize` 0_o .. made this change before realizing that i was debugging a "derp."

so this is for readability only, then.

File: cmd/datamon/cmd/cli_test.go [hunk 2]
```diff
@@ -307,6 +307,104 @@ func TestUploadBundle_filePath(t *testing.T) {
 	}, "observe error on bundle upload path as file rather than directory", true)
 }
 
+func getTempFile(t *testing.T, filename string) (*os.File, func()) {
```
Comment (fredbi) [inline cmd/datamon/cmd/cli_test.go]: nitpick/information sharing: on internal test functions, we may use `t testing.TB` interface instead. This allows test utility funcs to be reused by benchmarks.

File: pkg/core/bundle_unpack.go (hunk 1 above)
Comment (fredbi) [reply pkg/core/bundle_unpack.go]: :+1:

File: cmd/datamon/cmd/cli_test.go (hunk 2 above)
Comment (ransomw1c) [reply cmd/datamon/cmd/cli_test.go]: noted

**Pull request merged**